textproc/lilypond/Portfile

```diff
@@ -59,9 +61,72 @@ build.type          gnu
 configure.cmd       autoconf -f && ./configure
 configure.python    ${prefix}/bin/python3.8
 
+set lilypond.texgyredir \
+    "${prefix}/share/texmf-texlive/fonts/opentype/public/tex-gyre"
+set lilypond.mactex_bin   ""
+set lilypond.temp         ""
+set lilypond.have_texgyre false
+
+variant mactex description {Allow lilypond to use MacTeX or another\
+    external TeXLive installation instead of MacPorts TeXLive tools.  To\
+    install the 'mactex' variant the path to the TeX distribution's\
+    binary directory (for example '/Library/TeX/texbin') must be added\
+    to 'binpath' in 'macports.conf' *before* installing this port.} {
```

Comment:
This useful install hint (the 2nd sentence) should be moved into a `notes` block inside this variant.

Reply:
Hmm. My gut feeling says that the message should better stay in the `variant` output, since it is essential information that should be displayed *before* Joe User tries to install it.

Reply:
I don't see an immediate benefit of doubling the data, since the variant part contains checks that test whether MacTeX is actually installed and functioning.

Reply:
My issue is that the variant description never appears when installing the port. One would have to type `port variants lilypond` to see the note at all.

Reply:
hopefully the maintainer will chime in.

Reply:
This is a rather long message for `port variants`. I'd like to see a short message for the variant, with a longer note in `port notes`.

Reply:
I still don't get it.  What part of the message should become the 'longer note' that gets shown *after* installation?  All information currently in the variant message is intended to be shown *before* installation!  In comparison to, say, `port notes python37`, no user action has to be done after the package gets installed.

And why is it important to have a short variant message?

Reply:
Can you point me to another macports package that needs a prerequisite (probably with some additional explanation) before a variant can be installed, and that behaves as you envision?

Reply:
Users can invoke `port notes` anytime, just like `port variants`. In practice, one is a list of options and the other is open-ended notes.

I'm fairly certain that most Portfiles with a `prefetch` block in them try to abort builds as early as possible (perhaps due to missing dependencies).

Reply:
> Users can invoke `port notes` anytime, just like `port variants`. In practice, one is a list of options and the other is open-ended notes.

Thanks.  Unfortunately, this doesn't answer my questions.  Let me rephrase it: why shall I put stuff into `port notes` if the relevant information is (a) necessary only for a certain variant, and (b) only *before* installation?  Can you name a package in macports that behaves as you want me to change lilypond?

Reply:
A: it's _a lot_ of information. Too much even. Can you show that it's normal for such a large variant description?

B: `port notes` can be run prior to installation.

Reply:
`notes` are meant for installation information and variant descriptions are not. Most people won't think to look there if they have a problem and will simply file a ticket. Thus it falls in the maintainer's lap.

Reply:
There are some long variant descriptions in macports.  Examples: adblock2privoxy, calendar-contacts-server, macos-fortress.  And none of those packages have the variant information (or a variation of it) repeated in the `port notes` output.

Joe User will normally type `port variants lilypond`, and he will see there are only two variants.  Why is it important to have short and terse descriptions for just two options?  IMHO it's much better to have a detailed explanation than the opposite.

Interestingly, `port info` tells the user that there are variants, but it doesn't tell that there are notes...

And again my really serious questions: What exactly do you expect in the 'notes' section?  And how could the variant information be shortened?  I guess I'm dense, so please help.

Reply:
> `notes` are meant for installation information

This is not what `man portfile` says: "Notes for setup and use of the port. This is shown after the port is activated and anytime the `notes` command is used;" This clearly indicates that the 'notes' section is for information of what the user should do *after* installation.

> Most people won't think to look there if they have a problem and will simply file a ticket. Thus it falls in the maintainer's lap.

This is a good argument, so I could add
```
notes  "This port uses the 'mactex' variant
        and thus needs an external TeX installation."
```

or something similar to the 'mactex' variant block.  Would that work?

However, I still think that the current docstring of `variant` should not be shortened.

Reply:
I have to correct myself.  MacTeX is only a build dependency; in other words, after installing LilyPond it is not needed any more, and my suggested wording of a 'notes' block is wrong.  I'm really clueless now what I could write instead.  Please advise.